packit/api.py

```diff
@@ -362,6 +390,161 @@ def update_source_git(self, revision_range: str):
+                f"First unsynced commit in dist-git: {status.dist_git_range_start}."
+            )
+        elif status.source_git_range_start:
+            number_of_commits = len(
+                list(
+                    self.up.local_project.git_repo.iter_commits(
+                        f"{status.source_git_range_start}~..", ancestry_path=True
+                    )
+                )
+            )
+            click.echo(
+                f"'{source_git}' is ahead of '{dist_git}' by {number_of_commits} commits."
+            )
+            click.echo(
+                f'Use "packit-dev source-git update-dist-git {source_git} {dist_git}" '
+                f"to transform changes from '{source_git}' to '{dist_git}'."
+            )
+            click.echo(
+                f"First unsynced commit in source-git: {status.source_git_range_start}."
+            )
+        elif status.dist_git_range_start:
+            number_of_commits = len(
+                list(
+                    self.dg.local_project.git_repo.iter_commits(
+                        f"{status.dist_git_range_start}~..", ancestry_path=True
+                    )
+                )
+            )
+            short_hash = self.dg.local_project.git_repo.git.rev_parse(
+                status.dist_git_range_start, short=True
+            )
+            click.echo(
+                f"'{source_git}' is behind of '{dist_git}' by {number_of_commits} commits."
+            )
+            click.echo(
+                f'Use "packit-dev source-git update-source-git {dist_git} {source_git} '
+                f"{short_hash}~..\" to transform changes from '{dist_git}' to '{source_git}'."
+            )
+            click.echo(
+                f"First unsynced commit in dist-git: {status.dist_git_range_start}."
+            )
+        else:
+            click.echo(f"'{source_git}' is up to date with '{dist_git}'.")
+
+    def _get_latest_commit_update_pair(self) -> Tuple[str, str]:
+        """Finds the latest pair of commits which was created by updating
+        source-git from dist-git (or vice versa) denoted by git trailers.
+
+        Determining the latest pair by commit time is not sufficient since
+        the datetime may be the same in all commits. We need to check for
+        actual parent-children relationships to determine the ordering.
+
+        Returns:
+            A tuple containing hash of the source-git commit and hash of the
+                corresponding dist-git commit. This represents the latest
+                update pair in both source-git and dist-git.
+
+        Raises:
+            PackitException, if no commits with git trailers that are used
+                for checking the sync could be found.
+        """
+        # Use source-git commit first in both tuples for consistency
+        sg_update_commits = [
+            (
+                c.hexsha,
+                re.search(rf"{re.escape(FROM_DIST_GIT_TOKEN)}: (.+)", c.message).group(
```

Comment:
In order to avoid accidents :) 

```suggestion
                re.search(rf"^{re.escape(FROM_DIST_GIT_TOKEN)}: (.+)$", c.message).group(
```

Reply:
Also had to add `re.MULTILINE` in order for this to work